### ISSUE TITLE
Filter throw TypeInitializationException on net6.0 preview 3

### ISSUE THREAD

**Issue by aguacongas** [IharYakimush/comminity-data-odata-linq#35]
I try to filter EF entities with `Filter` with net6.0 but that throw a `TypeInitializationException` :

```cs
public static ODataQuery<T> GetODataQuery<T>(this IQueryable<T> query, string filter, IEdmModel edmModel = null) where T: class
{
       var odataQuery = query.OData(edmModel: edmModel);

       if (!string.IsNullOrEmpty(filter))
       {
            odataQuery = odataQuery.Filter(filter);
       }

       return odataQuery;
}
```

```bash
System.TypeInitializationException: The type initializer for 'Community.OData.Linq.OData.Query.Expressions.ExpressionBinderBase' threw an exception.
 ---> System.InvalidOperationException: Sequence contains more than one matching element
   at System.Linq.ThrowHelper.ThrowMoreThanOneMatchException()
   at System.Linq.Enumerable.TryGetSingle[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at Community.OData.Linq.OData.Query.Expressions.ExpressionBinderBase..cctor()
   --- End of inner exception stack trace ---
   at Community.OData.Linq.OData.Query.Expressions.ExpressionBinderBase.CreateBinaryExpression(BinaryOperatorKind binaryOperator, Expression left, Expression right, Boolean liftToNull)
   at Community.OData.Linq.OData.Query.Expressions.FilterBinder.BindBinaryOperatorNode(BinaryOperatorNode binaryOperatorNode)
   at Community.OData.Linq.OData.Query.Expressions.FilterBinder.Bind(QueryNode node)
   at Community.OData.Linq.OData.Query.Expressions.FilterBinder.BindExpression(SingleValueNode expression, RangeVariable rangeVariable, Type elementType)
   at Community.OData.Linq.OData.Query.Expressions.FilterBinder.BindFilterClause(FilterBinder binder, FilterClause filterClause, Type filterType)
   at Community.OData.Linq.OData.Query.Expressions.FilterBinder.Bind(IQueryable baseQuery, FilterClause filterClause, Type filterType, IServiceProvider requestContainer)
   at Community.OData.Linq.ODataLinqExtensions.Filter[T](ODataQuery`1 query, String filterText, String entitySetName)
```

```xml
<PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.0-preview.3.21201.2" />
<PackageReference Include="Community.OData.Linq.Json" Version="1.4.2" />
```

Same code works on net5.0.

**Comment by aguacongas** [comment]
Hi @IharYakimush , Are going to publish a preview nuget package soon including this fix ?

**Comment by IharYakimush** [comment]
Yes, @aguacongas I will do it today

**Comment by IharYakimush** [comment]
Released in version 1.4.3
FYI @aguacongas
